michaeldegroot/DoctrineEncryptBundle
# Ref #25 - Entity no longer always updated

This should fix issue #25.  

I based my PR on the code submitted in the issue by @benpic87 . I do welcome a proper validation of the code added as I'm not a specialist on doctrine-events.

### Summary of changes
Added pre-flush-test to see if old value and new value is the same and if so, keep the old encrypted value.

I added an `ArrayCollection` of previously decrypted entities to the subscriber, since `UnitOfWork` will recalculate the changes it needs to push before executing it's queries.  That's why all old encrypted values need to be re-encrypted before flushing.

### Additional unit-tests
2 unit-tests were added in `Tests/Functional/`: one for the `HaliteEncryptor`, one for the `DefuseEncryptor`.

For each decryptor three tests are added:
 * See if we can store the entity
 * See if we do not update the entity just by reading it
 * See if we store encrypted values in the database
